### PR TITLE
First commit of Phi In Jets task

### PR DESCRIPTION
**Pull request by smaff92**
First commit of a new analysis for Run 3 data, aimed to study resonances Phi and Kstar in/outside of jets

### CODE REVIEW

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:20]: Please fix this

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:185]: Please fix this

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Dear Vit,

This has now been fixed.
Best Regards
Adrian.N

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Dear Vit,

This has now been fixed.
Best Regards
Adrian.N

**Comment by vkucera** [comment]
Please do not force push during review.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:19]: You don't need these. Use `RecoDecay` features.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:16]: Put one author per line.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:51]: Move this up next to the other `Framework` includes.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:58]: Why do you need two registries?

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:62]: Underscores are only allowed in names of namespaces.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:192]: You use `abs` where it is useless in many places.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:264]:
```suggestion
    if (std::abs(lResonance.Rapidity()) > 0.5)
```

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:262]: All this can be done with `RecoDecay`.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:289]: All this can be done with `RecoDecay`.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [inline PWGJE/Tasks/PhiInJetsJE.cxx:248]: Names of functions and variables should start with small letters.

File: PWGJE/Tasks/CMakeLists.txt
Comment (vkucera) [inline PWGJE/Tasks/CMakeLists.txt:100]:
```suggestion
                        SOURCES phiInJets.cxx
```
Workflow files should start with small letters. The suffix `JE` is redundant.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Fixed

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: I prefer to use these for now, as they are familiar to me. I will update them once I can confirm that my code works on hyperloop.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Fixed

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Becasue I want one for LF tracks and one for JE tracks.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: fixed

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: I prefer to use these for now, as they are familiar to me. I will update them once I can confirm that my code works on hyperloop.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: I prefer to use these for now, as they are familiar to me. I will update them once I can confirm that my code works on hyperloop.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (smaff92) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: fixed

File: PWGJE/Tasks/CMakeLists.txt
Comment (smaff92) [reply PWGJE/Tasks/CMakeLists.txt]: fixed

**Comment by smaff92** [comment]
Dear Vit,

Unless you find any major errors in the code, could you please just approve the merge?
Then I will implement any minor suggestions for the next iteration of the code, for now I just want to test that I can run it on hyperloop.

Best Regards
Adrian.N

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: OK. But please come back to it. `TLorentzVector` is unnecessarily heavy here.

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: Why did you remove the other names? Are they not authors?

File: PWGJE/Tasks/PhiInJetsJE.cxx
Comment (vkucera) [reply PWGJE/Tasks/PhiInJetsJE.cxx]: But what difference does it make? All the histograms end up in the same folder anyway, no?

**Review by vkucera — APPROVED** [review]
Hi @smaff92 , thanks for addressing some of the comments. Please fix the remaining issues once you get your code running on Hyperloop.
You still need to fix the formatting by merging https://github.com/smaff92/O2Physics/pull/1 and approval from the code owners before your PR can be merged.

**Comment by ddobrigk** [comment]
@nzardosh @maoyx @mfasDa shall this be approved? Vit has agreed to merge as is pending future work. Please let me know - if no feedback arrives by the end of today I will approve from my side. Thanks!

**Review by ddobrigk — APPROVED** [review]
Approving since no reaction from approvers and reviewer is satisfied for now. Thanks!

**Comment by smaff92** [comment]
It seems that PWGHF/Core/PDG.h got removed two days ago?
I moved over to that header file upon Vit's first request.
For the sake of stability, I will now move back to <TDatabasePDG.h> and <TPDGCode.h>.

During the review process I have made major additions to the code, it now includes MC process functions as well (although the code data code has remained unchanged.)

**Comment by vkucera** [comment]
> It seems that PWGHF/Core/PDG.h got removed two days ago? I moved over to that header file upon Vit's first request. For the sake of stability, I will now move back to <TDatabasePDG.h> and <TPDGCode.h>.
> 
> During the review process I have made major additions to the code, it now includes MC process functions as well (although the code data code has remained unchanged.)

`PWGHF/Core/PDG.h` was replaced with [PhysicsConstants.h](https://github.com/AliceO2Group/AliceO2/blob/dev/Common/Constants/include/CommonConstants/PhysicsConstants.h). Please check the changes in https://github.com/AliceO2Group/O2Physics/pull/3976 for examples of how to modify your code accordingly. Do not revert back to the ROOT headers. Please follow the O2 announcement channel to stay up to date with the news.

**Comment by smaff92** [comment]
Dear @vkucera 

I have now changed the code to utilize PhysicsConstants.h
It compiles locally after a fresh git pull.

Could you please proceed with the test build?

Best Regards
Adrian.N

**Comment by vkucera** [comment]
Dear @smaff92 , thanks for the update.
The build tests run automatically. If this is your first PR in O2Physics, you will need an approval from the code owners to enable them. (Check the [contribution guidelines in the documentation](https://aliceo2group.github.io/analysis-framework/docs/gettingstarted/contributingtocode.html).)
In the meantime you have to fix the formatting to make your PR ready for merging.

**Comment by smaff92** [comment]
Dear @nzardosh, @maoyx, @mfasDa 

I just wanted to ping a request for approving the merge of my code into O2Physics.
If done today, I might be able to have results ready for the next PAG-JE meeting.
Thanks!

Best Regards
Adrian.N

**Review by nzardosh — APPROVED** [review]

**Pull request merged**
